client/src/featureform/serving.py

```diff
@@ -424,9 +446,129 @@ def __init__(self, stream, dataframe=None):
+            print(label)
+            clf.score(features, label)
+
+
+        # TRAIN OUTPUT
+        # np.array([
+        #   [1, 1, 3],
+        #   [5, 1, 2],
+        #   [7, 6, 5],
+        #   [8, 3, 3],
+        #   [5, 2, 2],
+        # ])
+        # np.array([2, 4, 2, 3, 4])
+        # np.array([
+        #   [3, 1, 2],
+        #   [5, 4, 5],
+        # ])
+        # np.array([6, 7])
+
+        # TEST OUTPUT
+        # np.array([
+        #   [5, 1, 3],
+        #   [4, 3, 1],
+        #   [6, 6, 7],
+        # ])
+        # np.array([4, 6, 7])
+
+        ```
+
+        Args:
+            test_size (float): The ratio of test set size to train set size. Must be a value between 0 and 1. If excluded
+                it will be the complement to the train_size. One of test_size or train_size must be specified.
+            train_size (float): The ratio of train set size to train set size. Must be a value between 0 and 1. If excluded
+                it will be the complement to the test_size. One of test_size or train_size must be specified.
+            shuffle (bool): Whether to shuffle the dataset before splitting.
+            random_state (Union[int, None]): A random state to shuffle the dataset. If None, the dataset will be shuffled
+                randomly on every call. If >0, the value will be used a seed to create random shuffle that can be repeated
+                if subsequent calls use the same seed.
+            batch_size (int): The size of the batch to return from the iterator. Must be greater than 0.
+
+        Returns:
+            train (Iterator): An iterator for training values.
+            test (Iterator): An iterator for testing values.
+        """
+        if batch_size < 1:
+            raise ValueError("batch_size must be 1 or greater")
+
+        if random_state == 0:
+            raise ValueError("random_state must be greater than zero or None")
+
+        test_size, train_size = self.validate_test_size(test_size, train_size)
+
+        name = self._stream.name
+        variant = self._stream.version
+        stub = self._stream._stub
+        model = self._stream.model if hasattr(self._stream, "model") else None
+        if random_state is None:
+            random_state = 0
+
+        train, test = TrainingSetTestSplit(
+            stub=stub,
+            name=name,
+            version=variant,
+            model=model,
+            test_size=test_size,
+            train_size=train_size,
+            shuffle=shuffle,
+            random_state=random_state,
+            batch_size=batch_size,
+        ).split()
+
+        return train, test
+
+    @staticmethod
+    def validate_test_size(test_size, train_size):
+        if test_size > 1 or test_size < 0:
+            raise ValueError("test_size must be between 0 and 1")
+        if train_size > 1 or train_size < 0:
+            raise ValueError("train_size must be between 0 and 1")
+        if test_size != 0 and train_size != 0:
```

Comment:
nit: you can remove this conditional by moving the nested conditional to the both at the first level. 

```
        if test_size == 0 and train_size != 0:
             test_size = 1 - train_size
         if test_size != 0 and train_size == 0:
             train_size = 1 - test_size

         if test_size + train_size != 1:
             raise ValueError("test_size + train_size must equal 1")

         return test_size, train_size
```